File: challenge-load-test-chat/test-by-locust/src/tasks/get_chat.py
from locust import task, TaskSet
from endpoints import Endpoints

class GetChatTask(TaskSet):
    @task
    def get_chat(self):
        chat_id = getattr(self.user, 'chat_id', None)
        if chat_id:
            self.client.get(Endpoints.CHAT_DETAIL.format(chat_id=chat_id))

File: challenge-load-test-chat/test-by-k6/src/tasks/get-chat.ts
import http from 'k6/http';
import { check } from 'k6';
import { Endpoints } from '../endpoints';

export function getChat(baseUrl: string, chatId: string) {
  const res = http.get(`${baseUrl}${Endpoints.CHAT_DETAIL(chatId)}`);
  check(res, { 'get chat status is 200': (r) => r.status === 200 });
  return res;
}
